src/app/util/Request/Request.js

```diff
@@ -180,3 +180,13 @@ export const listenForBroadCast = name => new Promise((resolve) => {
         };
     }
 });
+
+export const debounce = (callback, delay) => {
```

Comment:
Please explain the necessity of this in your code, why can not we manage without this, straightly speaking "hack"?